normandy/base/middleware.py

```diff
@@ -128,3 +136,109 @@ def process_request(self, request):
                 }
             )
         return response
+
+
+def default_username(email):
+    """Generate username for the Django user.
+
+    :arg str/unicode email: the email address to use to generate a username
+
+    :returns: str/unicode
+
+    """
+    # bluntly stolen from django-browserid
+    # store the username as a base64 encoded sha224 of the email address
+    # this protects against data leakage because usernames are often
+    # treated as public identifiers (so we can't use the email address).
+    return smart_text(base64.urlsafe_b64encode(
+        hashlib.sha1(force_bytes(email)).digest()
+    ).rstrip(b'='))
+
+
+class OIDCAccessTokenAuthenticationMiddleware:
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # XXX Here we *could* do something like:
+        # if not settings.OIDC_ACCESS_TOKEN_ENABLED:
+        #    from django.core.exceptions import MiddlewareNotUsed
+        #    raise MiddlewareNotUsed
+        self.oidc_user_endpoint = settings.OIDC_USER_ENDPOINT
+        self.UserModel = auth.get_user_model()
+
+    def __call__(self, request):
+        response = self.process_request(request)
+        if not response:
+            response = self.get_response(request)
+        return response
+
+    def process_request(self, request):
+        try:
+            access_token = re.findall(
+                'Bearer\s+(.*)',  # Is this right?
+                request.META.get('HTTP_AUTHORIZATION') or '',
+                re.I
+            )[0]
+        except IndexError:
+            # That's cool. If it's not there, it's not there.
+            # This middleware will not be able to authenticate the user
+            # for the duration of this request.
+            return
```

Comment:
In your snippet of code there you didn't use `re.IGNORE_CASE`. Was that deliberate?  The RFC doesn't appear to say it's case-insensitive. 
I think I'll steal this variant (instead of using `except IndexError`).

Reply:
Uh, no the lack of `re.IGNORE_CASE` was an accident. You should keep using that.